ctk/common/src/main/kotlin/org/codice/compliance/utils/report/Report.kt

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright (c) 2018 Codice Foundation
+
+Released under the GNU Lesser General Public License; see
+http://www.gnu.org/licenses/lgpl.html
+*/
+package org.codice.compliance.utils.report
+
+import org.codice.compliance.QUIET_REPORT
+import org.codice.compliance.SAMLComplianceException
+import org.fusesource.jansi.Ansi
+import org.junit.platform.engine.TestExecutionResult.Status.FAILED
+import org.junit.platform.engine.TestExecutionResult.Status.SUCCESSFUL
+
+object Report {
+
+    private val errorMessages = mutableMapOf<Sections, MutableList<SAMLComplianceException>?>()
+    private val coreHeaderSize = Sections.CORE_1.name.length
+
+    init {
+        Sections.values().forEach { errorMessages[it] = mutableListOf() }
+    }
+
+    private val reportQuietly by lazy {
+        System.getProperty(QUIET_REPORT) == "true"
+    }
+
+    fun addErrorMessage(section: Sections, error: SAMLComplianceException): Report {
+        // Error already added
+        if (errorMessages[section]?.isNotEmpty() == true &&
+                errorMessages[section]?.contains(error) == true) {
+            return this
+        }
+        errorMessages[section]?.add(error)
+        return this
+    }
+
+    fun getExceptionMessages(section: Sections): String? {
+        return errorMessages[section]?.joinToString()
+    }
+
+    fun hasErrors(): Boolean {
+        errorMessages.forEach {
+            if (it.value?.isNotEmpty() == true) {
+                return true
+            }
+        }
+        return false
+    }
+
+    @Suppress("StringLiteralDuplication", "NestedBlockDepth")
+    fun printReport() {
+        Sections.values().forEach {
+            //GENERAL
+            if (it == Sections.GENERAL) {
+                print(it.title)
+                print(it)
+            }
+
+            //SCHEMA
+            if (it == Sections.SCHEMA) {
+                print(it.title)
+                print(it)
+            }
+
+            // CORE
+            if (it.name.startsWith("CORE")) {
+                if (it == Sections.CORE_1) {
+                    println("CORE")
+                    println("\t${it.title}")
+                } else {
+                    if (it.name.length == coreHeaderSize) {
```

Comment:
❗️ This is really brittle. I think it would be better to enhance the enum so it carried some meta information about itself, info like `isTopLevel` and such. I've posted a gist with that and the rest of my suggested changes for this class here - https://gist.github.com/coyotesqrl/0a642eb17b4ab89cbe7e24a9d0277189

Reply:
Oh I see I can add multiple values to it 👌